src/theme/components/toggle.ts

```diff
@@ -0,0 +1,63 @@
+import { helperTextMargin } from "./global";
+
+// Style object for the Toggle's label
+const baseStyleLabel = {
+  fontSize: -1,
+  marginBottom: "0",
+  marginLeft: "xs",
+};
+
+// Style object for the Toggle's helper text
+const baseStyleHelper = {
+  ...helperTextMargin,
+  marginLeft: "xs",
+};
+
+const baseStyle = {
+  alignItems: "center",
+  label: baseStyleLabel,
+  helper: baseStyleHelper,
+};
+
+const Switch = {
+  baseStyle: {
+    opacity: 0.4,
+    track: {
+      p: "4px",
+      _checked: {
+        bg: "ui.link.primary",
+        opacity: 1,
+      },
+      _invalid: {
+        border: "1px solid",
+        borderColor: "ui.error.primary",
+        bg: "inherit",
+      },
+      _disabled: {
+        bg: "ui.gray.medium",
+        _checked: {
+          opacity: 0.4,
+        },
+      },
+    },
+    thumb: {
+      _disabled: {
+        bg: "ui.error.primary",
```

Comment:
Not sure why this isn't working?  Putting a color under `thumb` works

Reply:
Yeah, that's annoying. Maybe it doesn't pick this up directly.

Reply:
Oh wait, I think this should be `_invalid`. This is related to my other comment about the invalid state not displaying correctly.

Reply:
The `track` and `label` in Chakra's `Switch` have "invalid" style objects but the `thumb` does not... so what you can do is add a specific rule to the `track` object:

```
track: {
  ...
  _invalid: {
    border: "1px solid",
    borderColor: "ui.error.primary",
    bg: "inherit",
    "> span": {
      bg: "ui.error.primary",
    },
  },
}
```

Reply:
<img width="172" alt="Screen Shot 2021-12-06 at 5 57 54 PM" src="https://user-images.githubusercontent.com/1280564/144936269-3f7381d3-efe5-4a44-86ff-d3ebb2debece.png">

Reply:
how did you figure out that the "invalid" object was missing in "thumb" ?

Reply:
On the [Switch page](https://chakra-ui.com/docs/form/switch ), there is a link to the style source: https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/switch.ts

Typically, Chakra components should have this `_invalid` state available to them so I _think_ I just added it and it didn't work. If not that... then I probably saw that in the source code, the "data-invalid" attribute gets added to Chakra divs. That's a good indication of whether the underscore helper keys are available or not. If you try passing in `isRequired`, you should see `data-required`. It seems that's how Chakra keeps track of this. I could also be wrong and thinking of something else but I can't say I remember exactly how I came to the conclusion above.